File: Project Euler/Python/10001st prime.py
"""
    Topic : 
    Problem Statement :
                    By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
                        What is the 10 001st prime number?
"""
import itertools


def is_prime(n):
    for i in range(2, n//2 + 1):
        if n % i == 0:
            return False
        else:
            continue
    return True


p = 0
for x in itertools.count(1):
    if is_prime(x):
        if p == 10001:
            print(x)
            break
        p += 1
         

File: Project Euler/7. 10001st prime/10001st_prime.js
/* 

Topic : 10001st prime
Problem Statement : By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, 
                    we can see that the 6th prime is 13. What is the 10_001st prime number?
*/

function is_prime(num) {
  for (let i = 2; i < Math.sqrt(num) + 1; i++) {
    if (num % i === 0) {
      return false;
    }
  }
    return true;
}

while (true){
    let num = 0;
    let count = 0;
    while (count < 10001){
        num++;
        if (is_prime(num)){
        	count++;
        }
    }
    console.log(`${num} is the ${count}st prime number`);
    break;
}